src/cirq_iqm/iqm_sampler.py

```diff
@@ -180,3 +194,40 @@ def _send_circuits(
                 warnings.warn(f'Failed to abort job: {e}')
             finally:
                 sys.exit()
+
+
+@dataclass
+class ResultMetadata:
+    """Metadata for an IQM execution result.
+
+    Attributes:
+        job_id: A UUID representing the job.
+        calibration_set_id: A UUID representing the calibration set used for this result.
+        request: A RunRequest object representing the request made to run the circuit.
```

Comment:
```suggestion
        job_id: ID of the computational job.
        calibration_set_id: Calibration set used for this :class:`.IQMResult`.
        request: Request made to run the circuit.
```
The types should in general not be repeated in the docstrings, since they are already in the annotations and will show up in the docs.